xdsl/interactive/app.py

```diff
@@ -51,6 +56,46 @@
 ALL_PASSES = tuple(sorted((p_name, p()) for (p_name, p) in get_all_passes().items()))
 """Contains the list of xDSL passes."""
 
+ALL_PATTERNS = tuple(
+    (op_name, pattern_name, pattern)
+    for (op_name, pattern_by_name) in individual_rewrite.REWRITE_BY_NAMES.items()
+    for (pattern_name, pattern) in pattern_by_name.items()
+)
+"""Contains all the rewrite patterns."""
+
+
+def get_all_possible_rewrites(
+    patterns: tuple[tuple[str, str, RewritePattern], ...], op: ModuleOp
```

Comment:
I think introducing a couple of typealiases would make this code much more readable. In particular for `tuple[str, str, RewritePattern]`. Maybe something like `rewritedescrition` or something?

Reply:
`NamedTuple` might be most appropriate here